lib/cadence/worker.rb

```diff
@@ -66,18 +67,19 @@ def stop
 
     private
 
-    attr_reader :activities, :workflows, :pollers, :decision_middleware, :activity_middleware
+    attr_reader :options, :activities, :workflows, :pollers,
+                :decision_middleware, :activity_middleware
 
     def shutting_down?
       @shutting_down
     end
 
     def workflow_poller_for(domain, task_list, lookup)
-      Workflow::Poller.new(domain, task_list, lookup.freeze, decision_middleware)
+      Workflow::Poller.new(domain, task_list, lookup.freeze, decision_middleware, options)
```

Comment:
Should we avoid Worker Pollers and Activity Pollers taking the same lists of options?  We're passing in options that don't apply, like "thread_pool_size" sounds like it could apply to the Workflow Poller, but doesn't.  And we're not setting defaults in the same way.

Reply:
Agree. I plan to switch to initializing thread-pools within the worker and passing it down to the poller, not options